imsim/stamp.py

```diff
@@ -536,3 +581,165 @@ def draw(self, prof, image, method, offset, config, base, logger):
+                return image
+            obj = obj.evaluateAtWavelength(profile_wl)
+            obj = obj * self._trivial_sed
+        else:
+            self._fix_seds(obj, bandpass, logger)
+
+        # Normally, wcs is provided as an argument, rather than setting it directly here.
+        # However, there is a subtle bug in the ChromaticSum class where it can fail to
+        # apply the wcs correctly if the first component has zero realized flux in the band.
+        # This line sidesteps that bug.  And it never hurts, so even once we fix this in
+        # GalSim, it doesn't hurt to keep this here.
+        image.wcs = base['wcs']
+
+        if method == 'fft':
+            if self.fft_flux != self.nominal_flux:
+                obj = obj.withFlux(self.fft_flux, initial_flux_bandpass)
+
+            fft_image = image.copy()
+            fft_offset = offset
+            kwargs = dict(
+                method='fft',
+                offset=fft_offset,
+                image=fft_image,
+            )
+            if not faint and config.get('fft_photon_ops'):
+                kwargs.update({
+                    "photon_ops": galsim.config.BuildPhotonOps(config, 'fft_photon_ops', base, logger),
+                    "maxN": maxN,
+                    "rng": self.rng,
+                    "n_subsample": 1,
+                })
+
+            # Go back to a combined convolution for fft drawing.
+            prof = galsim.Convolve([obj] + psfs)
+            try:
+                fft_image = prof.drawImage(bandpass, **kwargs)
+            except galsim.errors.GalSimFFTSizeError as e:
+                # I think this shouldn't happen with the updates I made to how the image size
+                # is calculated, even for extremely bright things.  So it should be ok to
+                # just report what happened, give some extra information to diagonose the problem
+                # and raise the error.
+                logger.error('Caught error trying to draw using FFT:')
+                logger.error('%s',e)
+                logger.error('You may need to add a gsparams field with maximum_fft_size to')
+                logger.error('either the psf or gal field to allow larger FFTs.')
+                logger.info('prof = %r',prof)
+                logger.info('fft_image = %s',fft_image)
+                logger.info('offset = %r',offset)
+                raise
+            # Some pixels can end up negative from FFT numerics.  Just set them to 0.
+            fft_image.array[fft_image.array < 0] = 0.
+            if self.diffraction_fft:
+                self.diffraction_fft.apply(fft_image, bandpass.effective_wavelength)
+            fft_image.addNoise(galsim.PoissonNoise(rng=self.rng))
+            # In case we had to make a bigger image, just copy the part we need.
+            image += fft_image[image.bounds]
+            base['realized_flux'] = fft_image.added_flux
+        else:
+            obj = obj.withFlux(obj_cache.phot_flux, bandpass)
```

Comment:
At the moment, do_reweight for a given stamp is determined by whether the object has a precomputed flux and then whether it's in the same bandpass we're drawing. Is it guaranteed that if one object has a precomputed flux in a given bandpass, all others do as well? I'm trying to think how to determine do_reweight at the image rather than stamp level so the photon operator can work at the photon pool level.

Reply:
I doubt it's guaranteed, but it seems likely.  I think it's fine to just do the BandpassRatio here though.

Reply:
Great -- now done, inside the stamp as we discussed.